webapp/src/serial.tsx

```diff
@@ -0,0 +1,371 @@
+/// <reference path="../../localtypings/smoothie.d.ts" />
+
+import * as React from "react"
+import * as pkg from "./package"
+import * as core from "./core"
+import * as srceditor from "./srceditor"
+import * as sui from "./sui"
+import * as codecard from "./codecard"
+
+import Cloud = pxt.Cloud
+import Util = pxt.Util
+
+const lf = Util.lf
+
+export class Editor extends srceditor.Editor {
+    charts: Chart[] = []
+    chartIdx: number = 0
+    consoleBuffer: string = ""
+    // TODO pass these values in with props or config
+    isSim: boolean = true
+    maxConsoleLineLength: number = 500
+    maxConsoleEntries: number = 100
+    active: boolean = true
+    rawDataBuffer: string = ""
+    //TODO reasonable buffer size?
+    maxBufferLength: number = 5000
+
+    //refs
+    recordButton: HTMLElement
+    recordIcon: HTMLElement
+    consoleRoot: HTMLElement
+    chartRoot: HTMLElement
+
+    getId() {
+        return "serialEditor"
+    }
+
+    hasEditorToolbar() {
+        return false
+    }
+
+    setVisible(b: boolean) {
+        this.isVisible = b
+        if (b) this.startRecording()
+    }
+
+    acceptsFile(file: pkg.File) {
+        return file.name === pxt.SERIAL_EDITOR_FILE
+    }
+
+    isGraphable(v: string) {
+        return /[a-z]+:-?[0-9.]+/.test(v)
```

Comment:
does this match the regex used to extract log values?

Reply:
A suite of numbers would not be graphable.

Reply:
https://github.com/Microsoft/pxt/issues/3122